### ISSUE TITLE
Categorical Data

### ISSUE THREAD

**Issue by ccoroner** [nicodv/kmodes#5]
Hi Nico,

thank you very much for your work! I really appreciate your implementation and would like to use it for determining prototypes. Unfortunately I found it hard to follow the parameters of KPrototypes. For a future version I would wish for a documentation explaining what KMeans offers.

As a practical question: Is it possible to cluster on categorical data? Given the stocks example, how could I cluster the data based on the country and print out the chosen centroids?

Modifying the code to the following:
```python
syms = np.genfromtxt('stocks.csv', dtype=str, delimiter=',')[:, -1]    # changed index
X = np.genfromtxt('stocks.csv', dtype=object, delimiter=',')[:, :-1]   # changed index
kproto = kprototypes.KPrototypes(n_clusters=6, init='Huang', verbose=2)
clusters = kproto.fit_predict(X, categorical=[1, 2])
```

builds correctly syms and X:
```
syms = ['USA' 'USA' 'USA' 'USA' 'USA' 'USA' 'CN' 'USA' 'USA' 'USA' 'USA' 'NL']
X = [['AAPL' '738.5' 'tech']
 ['XOM' '369.5' 'nrg']
 ['GOOGL' '368.2' 'tech']
 ['MSFT' '346.7' 'tech']
 ['BRK-A' '343.5' 'fin']
 ['WFC' '282.4' 'fin']
 ['CHL' '282.1' 'tel']
 ['JNJ' '279.7' 'cons']
 ['WMT' '257.2' 'cons']
 ['VZ' '205.2' 'tel']
 ['ORCL' '192.1' 'tech']
 ['RDS-A' '195.7' 'nrg']]
```

but throws this error:
```
Traceback (most recent call last):
  File "stocks2.py", line 17, in <module>
    clusters = kproto.fit_predict(X, categorical=[1, 2])
  File "build/bdist.linux-x86_64/egg/kmodes/kmodes.py", line 325, in fit_predict
  File "build/bdist.linux-x86_64/egg/kmodes/kprototypes.py", line 339, in fit
  File "build/bdist.linux-x86_64/egg/kmodes/kprototypes.py", line 149, in k_prototypes
  File "build/bdist.linux-x86_64/egg/kmodes/kprototypes.py", line 42, in _split_num_cat
ValueError: could not convert string to float: RDS-A
```

Is there a parameter to make KPrototypes work on categorical data instead of float data?

**Comment by nicodv** [comment]
From the README:

> k-modes is used for clustering categorical variables. It defines clusters based on the number of matching categories between data points. (This is in contrast to the more well-known k-means algorithm, which clusters numerical data based on Euclidean distance.) The k-prototypes algorithm combines k-modes and k-means and is able to cluster mixed numerical / categorical data.

Your error is because you are applying an algorithm for mixed numerical / categorical data on categorical data. You should be using KModes instead.
